### PR TITLE
COMPAT: Support tz-aware timestamp types

### LINKED ISSUE #989: COMPAT: Support tz-aware timestamp types
```
In [1]: from ibis.impala.pandas_interop import pandas_to_ibis_schema

In [2]: pandas_to_ibis_schema(pd.DataFrame({'A': pd.date_range('20130101',periods=3,tz='US/Eastern')}))
IbisTypeError: Column A is dtype datetime64[ns, US/Eastern]
```

```
In [4]: ibis.__version__
Out[4]: 'v0.10.0+8.gac59450'

In [5]: pd.__version__
Out[5]: '0.19.2'
```

### PR DESCRIPTION
**Pull request by cpcloud**
This was a rabbit hole.

Closes #989

### CODE REVIEW

File: ibis/sql/alchemy.py [hunk 1]
```diff
@@ -82,7 +86,32 @@
...
+_numpy_dtype_map = {
+    # SQLAlchemyType: (non-nullable dtype, nullable dtype)
+    sa.SmallInteger: ('int16', 'float64'),
+    sa.SMALLINT: ('int16', 'float64'),
+    sa.Integer: ('int32', 'float64'),
+    sa.INTEGER: ('int32', 'float64'),
+    sa.BigInteger: ('int64', 'float64'),
+    sa.BIGINT: ('int64', 'float64'),
+    sa.Boolean: ('bool', 'object'),
+    sa.BOOLEAN: ('bool', 'object'),
```
Comment (wesm) [inline ibis/sql/alchemy.py]: Since this is the result of pandas peculiarities, does this introduce much technical debt?

Comment (cpcloud) [inline ibis/sql/alchemy.py]: It does tie to pandas missing value semantics which isn't great. Let me see if this is still necessary.

Comment (cpcloud) [inline ibis/sql/alchemy.py]: Turns out it wasn't necessary anymore. I removed it.

**Comment by cpcloud** [comment]
Merging on green.

**Pull request closed without merging**